http/src/main/resources/jupyter/init-actions.sh

```diff
@@ -348,17 +348,18 @@ END
 
       STEP_TIMINGS+=($(date +%s))
 
-      # If a Jupyter user script was specified, copy it into the jupyter docker container.
+      # If a Jupyter user script was specified, copy it into the jupyter docker container and execute it.
       if [ ! -z ${JUPYTER_USER_SCRIPT_URI} ] ; then
-        log 'Installing Jupyter user extension [$JUPYTER_USER_SCRIPT_URI]...'
+        log 'Running Jupyter user script [$JUPYTER_USER_SCRIPT_URI]...'
         gsutil cp ${JUPYTER_USER_SCRIPT_URI} /etc
         JUPYTER_USER_SCRIPT=`basename ${JUPYTER_USER_SCRIPT_URI}`
         docker cp /etc/${JUPYTER_USER_SCRIPT} ${JUPYTER_SERVER_NAME}:${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT}
         retry 3 docker exec -u root ${JUPYTER_SERVER_NAME} chmod +x ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT}
         # Execute the user script as privileged to allow for deeper customization of VM behavior, e.g. installing
         # network egress throttling. As docker is not a security layer, it is assumed that a determined attacker
         # can gain full access to the VM already, so using this flag is not a significant escalation.
-        docker exec --privileged -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT} &> us_output.txt || EXIT_CODE=$? && true ;
+        EXIT_CODE=0
+        docker exec --privileged -u root -e PIP_USER=false ${JUPYTER_SERVER_NAME} ${JUPYTER_HOME}/${JUPYTER_USER_SCRIPT} &> us_output.txt || EXIT_CODE=$?
```

Comment:
This seems correct, but I'm not sure if that `&& true` was added for any particular reason...

Reply:
The whole script is on `set -e`; presumably this was added to allow cluster start to succeed even if the init script fails

Reply:
Sorry - correction: it exits anyway if the script fails, but it wants to copy the log before exiting `init-actions.sh`, hence the `|| true`

Reply:
Yeah - it the current version will it copy the log even if the user script fails? I think so -- which is why I don't see why `&& true` was necessary.